driver-core/src/main/com/mongodb/operation/UserOperationHelper.java

```diff
@@ -27,21 +29,27 @@
 import org.bson.BsonValue;
 import org.bson.types.ObjectId;
 
-import java.util.Arrays;
+import java.util.Collections;
 
 import static com.mongodb.internal.authentication.NativeAuthenticationHelper.createAuthenticationHash;
 import static com.mongodb.operation.WriteConcernHelper.createWriteConcernError;
 import static com.mongodb.operation.WriteConcernHelper.hasWriteConcernError;
 
 final class UserOperationHelper {
+    private static final ServerVersion FOUR_ZERO = new ServerVersion(3, 7);
 
-    static BsonDocument asCommandDocument(final MongoCredential credential, final boolean readOnly, final String commandName) {
+    static BsonDocument asCommandDocument(final MongoCredential credential, final ConnectionDescription connectionDescription,
+                                          final boolean readOnly, final String commandName) {
+        boolean serverDigestPassword = connectionDescription.getServerVersion().compareTo(FOUR_ZERO) >= 0;
```

Comment:
Because we have no support for supported mechanisms, we default to the servers defaults (SCRAM-SHA-1, SCRAM-SHA-256).   SCRAM-SHA-256 requires undigested passwords.

This is a change in behaviour as in MongoDB 4.0 passwords are undigested.

Reply:
I'm not clear on the ramifications of this change.  What could this break for users?

Reply:
Ramifications are: Someone using the Create/Update UserOperation directly will now send a non hashed (digested) password in the document in 4.0.

Reply:
Ouch